src/deluge/storage/audio/audio_file_manager.cpp

```diff
@@ -322,17 +321,33 @@ Error AudioFileManager::getUnusedAudioRecordingFilePath(String* filePath, String
 		}
 	}
 
-	error = filePath->concatenate("/REC");
-	if (error != Error::NONE) {
-		return error;
-	}
-	error = filePath->concatenateInt(highestUsedAudioRecordingNumber[folderID], 5);
-	if (error != Error::NONE) {
-		return error;
+	// default to putting it in the normal spot if the song isn't named
+	if (songName->isEmpty()) {
+		error = filePath->concatenate("/REC");
+		if (error != Error::NONE) {
+			return error;
+		}
+		error = filePath->concatenateInt(highestUsedAudioRecordingNumber[folderID], 5);
+		if (error != Error::NONE) {
+			return error;
+		}
+		error = filePath->concatenate(".WAV");
+		if (error != Error::NONE) {
+			return error;
+		}
+		highestUsedAudioRecordingNumber[folderID]++;
 	}
-	error = filePath->concatenate(".WAV");
-	if (error != Error::NONE) {
-		return error;
+	else {
+		char namedPath[255]{0};
+		snprintf(namedPath, sizeof(namedPath), "%s/%s/%s.wav", filePath->get(), songName->get(), channelName);
+		int i = 1;
+		while (storageManager.fileExists(namedPath)) {
+			snprintf(namedPath, sizeof(namedPath), "%s/%s/%s_%d.wav", filePath->get(), songName->get(), channelName, i);
```

Comment:
```suggestion
			snprintf(namedPath, sizeof(namedPath), "%s/%s/%s_%03d.wav", filePath->get(), songName->get(), channelName, i);
```
Fixed-length format would make alphabetical sorting happier, at least until someone has over 1000 takes for a single instrument.